rfcs/0001-tokens-naming-convention.md

```diff
@@ -0,0 +1,495 @@
+
+For example if we don't use this specifier then we might end up creating a token like `color.background` for let's say a `Button` component. But this leads to an ambiguity since we we don't have any context attached to the color and we don't know what color we are referring to. 
+
+Rather if we start the token naming with a `Base` which means prefixing the component name then things might become more clear and contextual. For eg: `Button.color.background`
+
+But now if we want to refer a component within a component for example we need to refer to the `color` of an `icon` component inside a `Button` component then how shall we do that? An object can also split further and can have sub-component nesting to it as well. Eg: `Button.leftIcon.color.background` 
+
+Similarly, if we want to refer to a token value within a theme like `font.family` we can't do it alone since it doesn't has any context attached to it and we don't even know if what are we referring to? a theme token or a component token?
+
+But as soon as we attach the context our problem will be solved. For eg: `theme.font.family` tells us that we are trying to refer to font family token inside our theme.
+
+#### TL;DR
+**Object** refers to
+  1. **component** - `Button`, `theme` etc.
+  2. **sub-component** - one or more component within a component - `left-icon`, `right-icon` etc.
+
+### Base
+Base acts as a token’s backbone. Once we have `Object` as the first level classification the next level is `Base`. For example `color`, `font`, `size` etc becomes the `Base` i.e `Button.color`, `Button.font`, `theme.color` etc. 
+
+But as our collection grows the `Base` alone can't serve the purpose and it literally doesn't mean anything. For eg: `Button.color` itself is ambiguous becuase a color could be `textColor`, `backgroundColor`, `borderColor` or anything else, so instead we need a further level of pairing. Hence, the `Base` itself can be categorised into `category`, `property` and `behavior`.
+
+#### Category
+Categories span visual style concerns and may overlap at times
+
+Common categories includes but not limited to:
+- `color`
+- `font` (aka type, typography, text)
+- `space` (aka units, dimension, spacing)
+- `size` (aka sizing)
+- `elevation` (aka z-index, layer, layering)
+- `breakpoints` (aka media-query, responsive)
+- `shadow` (aka depth)
+- `time` (aka animation, duration)
+
+#### Property
+Property is something that attach some meaning to the category in such a way that a meaningful value can be derived
+
+Common properties includes but not limited to:
+- `text`
+- `background`
+- `border`
+- `weight`
+
+Usually a property is paired up with category to form a meaningful token value. For eg:
+- `color.text`
+- `color.background`
+- `color.border`
+- `font.weight`
+
+Some meaningful token name examples:
+```jsonc
+// Object.Base<category.property>
+Button.color.text
+Button.color.background 
+Button.color.border 
+Button.font.weight
+Button.font.size
+```
+
+#### Behavior
```

Comment:
I understand the need for this segmentation but `"behaviour"` with values `"heading"` `"body"` is slightly confusing.

Reply:
Yes and hence it's in the open question :)

Reply:
Yes, saw it later. 🙈